sqlair.go

```diff
@@ -163,3 +164,61 @@ func (q *Query) One(outputArgs ...any) error {
 	iter.Decode(outputArgs...)
 	return iter.Close()
 }
+
+// All iterates over the query and decodes all the rows.
+// resultColumns must contain pointers to slices of each of the query arguments e.g.
+// > q.All(&[]*Person{}, &[]*Address{}) or q.All(&[]Address{}, &[]Person{})
+// It fabricates all structs as needed.
+func (q *Query) All(slicePointers ...any) error {
+	if q.err != nil {
+		return q.err
+	}
+
+	// Check slice inputs
+	var slicePtrVals = []reflect.Value{}
+	var sliceVals = []reflect.Value{}
+	for _, ptr := range slicePointers {
+		ptrVal := reflect.ValueOf(ptr)
+		if ptrVal.Kind() != reflect.Pointer {
+			return fmt.Errorf("need pointer to slice, got %s", ptrVal.Kind())
```

Comment:
I'm not sure about how we phrase this in Decode, but maybe something like this would be helpful:

`cannot decode result: need ...`

Otherwise this error might be a bit mysterious depending on where it pops up.

Same for the similar cases below.